addons/stock/models/stock_scrap.py

```diff
@@ -171,6 +171,8 @@ def action_get_stock_move_lines(self):
 
     def action_validate(self):
         self.ensure_one()
+        if self.scrap_qty <= 0:
```

Comment:
```suggestion
        if fields.Float.float_is_zero(self.scrap_qty, precision_roundingself.product_uom_id.rounding):
```
To correctly handle UoM

Reply:
@amoyaux thanks for the suggestion, updated the pr with the given suggestion